src/core/weightedGraph.test.js

```diff
@@ -60,4 +61,37 @@ describe("core/weightedGraph", () => {
       expectEqual(wg, wg_);
     });
   });
+
+  describe("overrideWeights", () => {
+    const example = () => {
+      const graph = new Graph().addNode(foo).addNode(bar);
+      const weights = Weights.empty();
+      weights.nodeWeights.set(foo.address, 1);
+      weights.edgeWeights.set(foobar.address, {forwards: 2, backwards: 2});
+      weights.edgeWeights.set(EdgeAddress.empty, {forwards: 2, backwards: 2});
+      return {graph, weights};
+    };
+    it("has no effect if the overrides are empty", () => {
+      const g1 = example();
+      const g2 = WeightedGraph.overrideWeights(g1, Weights.empty());
+      expectEqual(g1, g2);
+    });
+    it("takes weights from base and overrides, choosing overrides on conflicts", () => {
+      const overrides = Weights.empty();
+      overrides.nodeWeights.set(foo.address, 100);
+      overrides.nodeWeights.set(bar.address, 100);
+      overrides.edgeWeights.set(foobar.address, {
+        forwards: 100,
+        backwards: 100,
+      });
```

Comment:
Nit: same here, think it would be better to have all the weight numbers used be distinct.

Reply:
Done.